File: beginner/2.globalview/xmp_init_ans.f90
program init
  include 'xmp_lib.h'
!$xmp nodes p(2)
!$xmp template t(10)
!$xmp distribute t(block) onto p
  integer :: a(10) 
!$xmp align a(i) with t(i)
  integer :: i

!$xmp loop on t(i)
  do i=1,10
    a(i)=i 
  end do

!$xmp loop on t(i)
  do i=1,10
    print *, xmp_node_num(), a(i)
  end do

end program init

File: beginner/2.globalview/xmp_init_ans.c
#include <stdio.h>
#include <xmp.h>
#pragma xmp nodes p[2]
#pragma xmp template t[10]
#pragma xmp distribute t[block] onto p
int a[10]; 
#pragma xmp align a[i] with t[i]

int main(){
  
#pragma xmp loop on t[i]
  for(int i=0;i<10;i++)
    a[i] = i+1; 

#pragma xmp loop on t[i]
  for(int i=0;i<10;i++)
    printf("[%d] %d\n", xmpc_node_num(), a[i]);

  return 0;
}
